src/ott/problems/nn/dataset.py

```diff
@@ -0,0 +1,155 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Neural OT datasets."""
+
+import dataclasses
+from typing import Any, Iterable, Iterator, Literal, NamedTuple, Tuple
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+__all__ = ['create', 'Dataloaders', 'GaussianMixture']
+
+Name_t = Literal['gaussian_simple', 'gaussian_circle', 'gaussian_square_five',
+                 'gaussian_square_four']
+
+
+class Dataloaders(NamedTuple):
+  r"""Train and validation dataloaders for the source and target measures.
+
+  Args:
+  trainloader_source: Training dataset, source measure
+  trainloader_target: Training dataset, target measure
+  validloader_source: Valid dataset, source measure
+  validloader_target: Valid dataset, target measure
+  """
+  trainloader_source: Iterable[jnp.ndarray]
+  trainloader_target: Iterable[jnp.ndarray]
+  validloader_source: Iterable[jnp.ndarray]
+  validloader_target: Iterable[jnp.ndarray]
+
+
+GAUSSIAN_CENTERS = {
```

Comment:
I don't think this really needs a global constant, would just move to `__post_init__`.